src/engine/controls/cuecontrol.cpp

```diff
@@ -2168,11 +2172,72 @@ void CueControl::updateIndicators() {
             m_pCueIndicator->setBlinkValue(ControlIndicator::ON);
         }
     }
+
+    // Update hotcue indicators: activate the indicator if we are at the cue pos
+    // or if we passed it since the last update, else deactivate.
+    auto prevPos = m_prevPosition.getValue();
+    auto currPos = frameInfo().currentPosition;
+    VERIFY_OR_DEBUG_ASSERT(prevPos.isValid()) {
+        m_prevPosition.setValue(currPos);
+        return;
+    }
+
+    // These are used to check whether we passed the cue while the play position
+    // was wrapped around (loop of full-track repeat).
+    // Note: notifyWrapAround() guarantees that in/out are valid if count > 0.
+    auto loopOut = m_jumpPos;
+    auto loopIn = m_targetPos;
+    bool jump = m_wrapAroundCount == 1;
```

Comment:
I do not understand the "jump" Issue. Why is that only exact 1 and not 2?

Reply:
By "jump" I mean one regular seek loop out -> in. Usually the buffered samples don't cover the entire loop. In case of a very short loop the current position _might_ be after the previous position (= overlap of [in->curr] and [prev-out]), but we don't know for sure (unless we don another comparison, curr > prev¹). 
With count 2 we know for sure that the entire loop range was read and therefore it suffices to check whether the cue is inside the loop.

¹I'll check if that makes the situation even easier.

Reply:
> ¹I'll check if that makes the situation even easier.

Yes, another check to detect `fullLoop` (one jump + overlap case) avoids the two comparisons for that case _per cue_.
But the code needs more explanation then (IMO).

Please check with the last commit if that makes sense.